### PR TITLE
Extend CommandDetailParser to include ACL details

### PR DESCRIPTION
**Pull request by sokomishalov**

### CODE REVIEW

File: src/main/java/io/lettuce/core/models/command/CommandDetailParser.java [hunk 1]
```diff
@@ -119,6 +147,22 @@ private static CommandDetail parseCommandDetail(Collection<?> collection) {
...
     }
 
+    private static Set<CommandDetail.AclCategory> parseAclCategories(Object aclCategories) {
+        Set<CommandDetail.AclCategory> result = new HashSet<>();
+
+        if (aclCategories instanceof Collection<?>) {
+            Collection<?> collection = (Collection<?>) aclCategories;
+            for (Object o : collection) {
+                CommandDetail.AclCategory aclCategory = ACL_CATEGORY_MAPPING.get(o);
+                if (aclCategory != null) {
```
Comment (mp911de) [inline src/main/java/io/lettuce/core/models/command/CommandDetailParser.java:157]: Should we also consider future extensions of ACL categories? What if Redis reports a new one, that isn't known to our Client?

Comment (sokomishalov) [reply src/main/java/io/lettuce/core/models/command/CommandDetailParser.java:157]: Actually, I think we should.
However, I decided to use the same "enum" approach (like `CommandDetail.Flag`).
I do think it is not flexible at all - neither `CommandDetail.Flag`, nor `CommandDetail.AclCategory`. Each of them is subject to change any day.
Using plain strings would be more flexible for both data structures.

Comment (mp911de) [reply src/main/java/io/lettuce/core/models/command/CommandDetailParser.java:157]: Hum, you're right, we have similar behavior in other places. So let's stick with the code as-is. Once we face changes in that area, we can make the approach more extensible.

Comment (sokomishalov) [reply src/main/java/io/lettuce/core/models/command/CommandDetailParser.java:157]: I agree - we don't know yet how often these flags and categories will be added.

**Comment by mp911de** [comment]
Thank you for your contribution. That's merged and polished now.

**Pull request closed without merging**